### PR TITLE
Fix Issue 17752 - Switch skips over declaration issued for explicitly uninitialized variables

### PR DESCRIPTION
**Pull request by wilzbach**
Required for https://github.com/dlang/dmd/pull/7817

### CODE REVIEW

**Comment by mathias-lang-sociomantic** [comment]
Any chance you could document `checkVar` in the process, most notably its return value ?

**Comment by mathias-lang-sociomantic** [comment]
Also, should this target stable ?

**Comment by wilzbach** [comment]
> Any chance you could document checkVar in the process, most notably its return value ?

I wanted to do another PR to change the deprecation message, so yeah. Why not ;-)

> Also, should this target stable ?

Per policy only "critical", "blocker" and "regression" should be on stable, but sure I don't see any reason why it shouldn't (rebased).

**Review by mathias-lang-sociomantic — APPROVED** [review]

**Comment by mathias-lang-sociomantic** [comment]
> I wanted to do another PR to change the deprecation message, so yeah. Why not ;-)

Fine by me if you want it in another PR

**Comment by wilzbach** [comment]
> Fine by me if you want it in another PR

https://github.com/dlang/dmd/pull/7844

> The Travis CI build failed

FYI: This failure is unrelated. We already have removed Travis on the master branch.

**Review by JinShil — DISMISSED** [review]

File: src/dmd/statement.d [hunk 1]
```diff
@@ -1605,7 +1605,9 @@ extern (C++) final class SwitchStatement : Statement
             }
             else
             {
-                deprecation("'switch' skips declaration of variable %s at %s", vd.toPrettyChars(), vd.loc.toChars());
+                if (!vd._init.isVoidInitializer)
+                    deprecation("'switch' skips declaration of variable %s at %s", vd.toPrettyChars(), vd.loc.toChars());
```
[the remark this answers is not in the document]
Comment (wilzbach) [reply src/dmd/statement.d]: This is already done in master, this here  is the stable branch. Doing such redundant work is pointless and will create extra work and conflicts for stable/master merges.

[the remark this answers is not in the document]
Comment (JinShil) [reply src/dmd/statement.d]: Ok, I understand.

**Review by JinShil — APPROVED** [review]

File: test/compilable/test17752.d [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// REQUIRED_ARGS: -c -de
```
Comment (WalterBright) [inline test/compilable/test17752.d]: Generally, please put a link to the bugzilla issue that is addressed.

Comment (wilzbach) [reply test/compilable/test17752.d]: Ok done.

**Pull request merged**